compiler/rustc_hir_analysis/src/collect/item_bounds.rs

```diff
@@ -95,6 +95,7 @@ pub(super) fn explicit_item_bounds(
             span,
             ..
         }) => opaque_type_bounds(tcx, def_id, bounds, *span, *in_trait),
+        hir::Node::Item(hir::Item { kind: hir::ItemKind::TyAlias(..), .. }) => &[],
```

Comment:
do we not need to return the bounds on the type alias here, or well, return them if the alias should be treated as an actual type?

Reply:
for `opaque` types these are the bounds like `impl These + Bounds`, for projections these are the bounds on the associated type declaration, for weak type aliases there is no equivalent. The bounds on generic params and the where bounds are handled in `predicates_of`